src/viking.js

```diff
@@ -1,11 +1,88 @@
 // Soldier
-class Soldier {}
+
+class Soldier {
+    constructor(health, strength) {
+      this.health = health,
+      this.strength = strength
+    }
+    attack() {
+     return this.strength
+    }
+    receiveDamage(damage) {
+      this.health = this.health - damage
+      
+    }
+  }
 
 // Viking
-class Viking {}
+
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+      super(health,strength)
+      this.name = name
+   }
+   battleCry() {
+   return ("Odin Owns You All!")
+   }
+   receiveDamage(damage) {
+   this.health = this.health - damage
+     if (this.health > 0) {
+        return (`${this.name} has received ${damage} points of damage`)
+     } else {
+     
+       return (`${this.name} has died in act of combat`)
+     }
+   }
+}
 
 // Saxon
-class Saxon {}
+
+class Saxon extends Soldier {
+
+    receiveDamage(damage) {
+      this.health = this.health - damage
+      if (this.health > 0) {
+        return (`A Saxon has received ${damage} points of damage`)
+      } else {
+        
+        return ("A Saxon has died in combat")
+      }
+    }
+  }
 
 // War
-class War {}
+
+class War {
+    constructor () {
+      this.vikingArmy = [],
+        this.saxonArmy = []  
+    }
+    addViking(viking) {
+      this.vikingArmy.push(viking)
+    }
+    addSaxon(saxon) {
+      this.saxonArmy.push(saxon)
+    }
+  
+    vikingAttack() {
+      let  randomsaxon = Math.floor(Math.random() * this.saxonArmy.length)
+      let  randomviking = Math.floor(Math.random() * this.vikingArmy.length)
+      return this.saxonArmy[randomsaxon].receiveDamage(this.vikingArmy[randomviking].attack())
```

Comment:
`let attackResult = this.saxonArmy[randomsaxon].receiveDamage(this.vikingArmy[randomviking].attack())`